### ISSUE TITLE
filter_vep --only_matched is not working as it should

### ISSUE THREAD

**Issue by PedroBarbosa** [Ensembl/ensembl-vep#644]
Hi,

I want to keep blocks of consequences belonging to a single gene, but apparently VEP is not removing the unwanted genes.

- VEP version: 97
- VEP Cache version: 97

CMD:
```
filter_vep --filter "SYMBOL is MYBPC3" --vcf_info_field ANN -i variants.vcf.bgz -o mypbc3.blocks.vcf--force_overwrite --only_matched
```
Example input variant:

```
11	47371624	MYBPC3:c.446C>A	G	T	.	.	ANN=T|downstream_gene_variant|MODIFIER|SPI1|ENSG00000066336|Transcript|ENST00000227163|protein_coding||||||||||rs779493486|4840|-1||SNV|HGNC|11241|||||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.90||||||||13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000256993|protein_coding|4/33||ENST00000256993.4:c.446C>A|ENSP00000256993.4:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.21)|benign(0.13)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||neutral(0.460)|Neutral(0.759)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|downstream_gene_variant|MODIFIER|SPI1|ENSG00000066336|Transcript|ENST00000378538|protein_coding||||||||||rs779493486|4787|-1||SNV|HGNC|11241|||||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.90||||||||13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000399249|protein_coding|4/34||ENST00000399249.2:c.446C>A|ENSP00000382193.2:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.21)|benign(0.196)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||deleterious(0.493)|Neutral(0.752)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant&NMD_transcript_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000544791|nonsense_mediated_decay|4/27||ENST00000544791.1:c.446C>A|ENSP00000444259.1:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.15)|benign(0.311)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||deleterious(0.581)|Neutral(0.822)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000545968|protein_coding|4/35||ENST00000545968.1:c.446C>A|ENSP00000442795.1:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.21)|benign(0.13)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||neutral(0.460)|Neutral(0.759)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00001003551|promoter||||||||||rs779493486||||SNV|||||||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||||||||||13.11|0.974678||||||||||||||||||||||||||||||3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0
```

output variant:
```
11	47371624	MYBPC3:c.446C>A	G	T	.	.	ANN=T|downstream_gene_variant|MODIFIER|SPI1|ENSG00000066336|Transcript|ENST00000227163|protein_coding||||||||||rs779493486|4840|-1||SNV|HGNC|11241|||||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.90||||||||13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000256993|protein_coding|4/33||ENST00000256993.4:c.446C>A|ENSP00000256993.4:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.21)|benign(0.13)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||neutral(0.460)|Neutral(0.759)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|downstream_gene_variant|MODIFIER|SPI1|ENSG00000066336|Transcript|ENST00000378538|protein_coding||||||||||rs779493486|4787|-1||SNV|HGNC|11241|||||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.90||||||||13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000399249|protein_coding|4/34||ENST00000399249.2:c.446C>A|ENSP00000382193.2:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.21)|benign(0.196)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||deleterious(0.493)|Neutral(0.752)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant&NMD_transcript_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000544791|nonsense_mediated_decay|4/27||ENST00000544791.1:c.446C>A|ENSP00000444259.1:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.15)|benign(0.311)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||deleterious(0.581)|Neutral(0.822)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|missense_variant|MODERATE|MYBPC3|ENSG00000134571|Transcript|ENST00000545968|protein_coding|4/35||ENST00000545968.1:c.446C>A|ENSP00000442795.1:p.Ala149Asp|501|446|149|A/D|gCc/gAc|rs779493486||-1||SNV|HGNC|7551||tolerated(0.21)|benign(0.13)||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||0.00||||||neutral(0.460)|Neutral(0.759)|13.11|0.974678|10.7507|18.06|0.96085648470747187|1.278436|1.235919|0.31&0.38&0.32|3.43||||0.04967|T|0.0894|T|-1.0557|0.301|0.55&.&.|1&1&1|-0.16&-0.16&-0.16|0.504&.&.|0.13&.&.|0.078|0.27&0.254&0.254|0.164&0.177&0.176|0.415&0.384&0.369|0.62565|0.403107|0.003000|0.230000|3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0,T|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00001003551|promoter||||||||||rs779493486||||SNV|||||||11:g.47371624G>T|||||||||5.866e-06|0|0|0|0|0|0|0|4.271e-05|4.271e-05|gnomAD_SAS||||18414213|||||||||||||||13.11|0.974678||||||||||||||||||||||||||||||3.4300000667572|3.4300000667572|2.69199991226196|0.814000010490417|rs779493486|3.18552e-05|0|0;ReMM=0.8250;DANN=0.961;GERP=3.43;phastCons=0.814;phyloP=2.692;fitcons=0.4031;SiPhy=10.7507;fathmmMKL=0.62565;Eigen=1.235919;Eigen-PC=1.278436;traP=0.001;dpsi_max_tissue=0.3982;dpsi_zscore=1.019
```

Thanks in advance,
Pedro

**Comment by aparton** [comment]
Hi Pedro,

I suspect this issue is caused by an issue with the header lines in your VCF file. Could you please send me the full VEP command that you used to generate your input line?

Kind Regards,
Andrew

**Comment by PedroBarbosa** [comment]
Hi @aparton ,

Please find my VEP command below:

```
srun shifter -V=/mnt/nfs/lobo/IMM-NFS/ensembl_vep:/media --image=ensemblorg/ensembl-vep:latest vep --force_overwrite --stats_text --hgvs --hgvsg --pubmed --check_existing --cache --dir /media/cache  --sift b --polyphen b --numbers --regulatory --variant_class --cache_version 97 -a GRCh37 --fasta /media/cache/homo_sapiens/97_GRCh37/Homo_sapiens.GRCh37.75.dna.primary_assembly.fa.gz --plugin dbscSNV,/media/custom_data/dbscSNV/hg19/dbscSNV1.1_hg19.txt.gz --plugin ExACpLI,/media/custom_data/ExACpLI/ExACpLI_values.txt --plugin GeneSplicer,/media/custom_data/genesplicer/bin/linux/genesplicer,/media/custom_data/genesplicer/human --plugin MaxEntScan,/media/custom_data/MaxEntScan --plugin SpliceRegion --plugin Condel,/media/custom_data/condel,b --plugin Carol --plugin CADD,/media/custom_data/cadd/hg19/v1.4/whole_genome_SNVs.tsv.gz,/media/custom_data/cadd/hg19/v1.4/InDels.tsv.gz --plugin dbNSFP,'consequence=ALL',/media/custom_data/dbNSFP/hg19/dbNSFP4.0b1a_hg19.txt.gz,GERP++_RS,phyloP30way_mammalian,29way_logOdds,phastCons30way_mammalian,MutationAssessor_score,SIFT4G_score,SIFT_score,Polyphen2_HDIV_score,Polyphen2_HVAR_score,MutationTaster_score,FATHMM_score,fathmm-MKL_coding_score,PROVEAN_score,CADD_phred,DANN_score,Eigen-pred_coding,Eigen-PC-phred_coding,MetaSVM_score,MetaSVM_pred,MetaLR_score,MetaLR_pred,REVEL_score,integrated_fitCons_score,LRT_score,LRT_pred,MutPred_score,VEST4_score,M-CAP_score,LINSIGHT --custom /media/custom_data/gerp/All_hg19_RS.bw,GERP_vep,bigwig,exact,0 --custom /media/custom_data/gerp/All_hg19_RS.bw,GERP_vep_all,bigwig,overlap,0 --custom /media/custom_data/phyloP100/hg19.100way.phyloP100way.bw,phyloP_vep,bigwig,exact,0 --custom /media/custom_data/phastcons100/hg19.100way.phastCons.bw,phastCons_vep,bigwig,exact,0 --custom /media/custom_data/gnomAD/hg19/gnomAD_v2.1_justImportantFields.vcf.gz,gnomADg,vcf,exact,0,AF,AF_nfe,AF_nfe_nwe --af --af_1kg --af_esp  --max_af --af_gnomad --pick --port 3337 -i /mnt/nfs/lobo/MCFONSECA-NFS/pedro.barbosa/resources/santaMaria_patients_sandraCheck/variant_ids.txt --vcf --vcf_info_field ANN --compress_output bgzip --output_file variants.vcf.bgz
```

Best,
Pedro

**Comment by aparton** [comment]
Hi Pedro,

Thanks for this. I've taken a look this afternoon and I'm unable to reproduce this issue. I suspect that it's an issue with your header lines. Would it be possible for you to send me example VEP output that causes this filter issue with the full header lines included?

Kind Regards,
Andrew

**Comment by PedroBarbosa** [comment]
Hi @aparton 

I just looked again at this issue. I run VEP again (latest docker image) and apparently the issue was solved. Now when I filter by gene name, only the right blocks of consequences are kept, as expected.

I'll close the issue for now. I'll reopen in the future if necessary.
Best,
Pedro

**Comment by PedroBarbosa** [comment]
Hi,

I need to reopen this issue since I keep struggling with filtering consequences. I can't see what does the header has to do with the problem. 

Instead of using filter_vep script I tried now apply the `--transcript_filter`option while running vep. In my case I selected `--per_gene --pick_order canonical --transcript_filter "SYMBOL is MYBPC3"`

All that I get is intergenic consequences. I can't get it.
Best regards,
Pedro
